agatha/construct/semrep_util.py

```diff
@@ -302,15 +318,101 @@ def _get_flags(self, input_path:Path, output_path:Path)->List[str]:
     if self.word_sense_disambiguation:
       res.append("--word_sense_disambiguation")
     res.append("--xml_output_format")
-    res.append(str(input_path))
-    res.append(str(output_path))
+    if input_path is not None and output path is not None:
+      res.append(str(input_path))
+      res.append(str(output_path))
+    else:
+      raise ValueError("Both input and output paths must be set or None")
     return res
 
-  def run(self, input_path:Path, output_path:Path)->None:
+  def _read_lines(self, in_file:Path)->List[str]:
+    """
+    Reads in lines of a file in memory.
+
+    Args:
+     in_file: Path to input file
+    Returns:
+      lines in the files
+    """
+    with open(in_file, 'r') as f:
+      lines = []
+      for line in f:
+        lines.append(line.strip())
+    return lines
+
+  def _write_xml(self, data: List[lxml.etree._Element], out_file:Path)->None:
+    """
+    Writes the produced XML to the specified
+    file.
+
+    Args:
+      data: List of xml elements produced by semrep
+      out_file: Path to output xml file
+    """
+    assert out_file.suffix == '.xml', 'The output must be a valid XML file'
```

Comment:
Before you open a file, please always `assert not out_file.exists()`
Having lost literally millions of data files to incorrectly configured file paths, we want to break instantly if we detect some unexpected behavior.